controllers/managedclusterview_controller.go

```diff
@@ -0,0 +1,186 @@
+	logger := r.Logger.With("ManagedClusterView", req.NamespacedName)
+	logger.Info("Reconciling ManagedClusterView")
+
+	var managedClusterView viewv1beta1.ManagedClusterView
+	if err := r.Client.Get(ctx, req.NamespacedName, &managedClusterView); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			logger.Error("Failed to get ManagedClusterView", "error", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := createOrUpdateConfigMap(ctx, r.Client, managedClusterView, r.Logger); err != nil {
+		logger.Error("Failed to create or update ConfigMap for ManagedClusterView", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully reconciled ManagedClusterView", "name", managedClusterView.Name)
+
+	return ctrl.Result{}, nil
+}
+
+func createOrUpdateConfigMap(ctx context.Context, c client.Client, managedClusterView viewv1beta1.ManagedClusterView, logger *slog.Logger) error {
+	logger = logger.With("ManagedClusterView", managedClusterView.Name, "Namespace", managedClusterView.Namespace)
+
+	var resultData map[string]string
+	err := json.Unmarshal(managedClusterView.Status.Result.Raw, &resultData)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal result data: %v", err)
+	}
+
+	reverseLookup := make(map[string]string)
+
+	for key, value := range resultData {
+		if strings.HasSuffix(key, ".config.yaml") {
+			var config StorageClusterConfig
+			err := json.Unmarshal([]byte(value), &config)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal config data for key %s: %v", key, err)
+			}
+
+			providerInfo := config
+			providerInfo.Clients = nil
+			providerInfo.ProviderClusterName = managedClusterView.Namespace
+
+			providerInfoJSON, err := json.Marshal(providerInfo)
+			if err != nil {
+				return fmt.Errorf("failed to marshal provider info: %v", err)
+			}
+
+			for _, client := range config.Clients {
+				reverseLookup[client] = string(providerInfoJSON)
+			}
+		}
+	}
+
+	configMapData := make(map[string]string)
+	for client, info := range reverseLookup {
+		configMapData[client] = info
+	}
+
+	operatorNamespace := os.Getenv("POD_NAMESPACE")
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ClientInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+		Data: configMapData,
+	}
+
+	op, err := controllerutil.CreateOrUpdate(ctx, c, configMap, func() error {
+		configMap.Data = configMapData
+
+		ownerExists := false
+		for _, ownerRef := range configMap.OwnerReferences {
+			if ownerRef.UID == managedClusterView.UID {
+				ownerExists = true
+				break
+			}
+		}
+
+		if !ownerExists {
+			ownerRef := *metav1.NewControllerRef(&managedClusterView, viewv1beta1.GroupVersion.WithKind("ManagedClusterView"))
+			logger.Info("OwnerRef added", "UID", string(ownerRef.UID))
+			configMap.OwnerReferences = append(configMap.OwnerReferences, ownerRef)
+		}
+		return nil
```

Comment:
Maybe we can replace it with [controller-util](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetOwnerReference) to make it simpler?

Reply:
can we use `c.Scheme()` ?

Reply:
I used it but it was not getting set somehow. The tests were failing. I will look into this later